app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -223,13 +223,39 @@ def associate
         if @compute_resource.supports_host_association?
           associator = ComputeResourceHostAssociator.new(@compute_resource)
           associator.associate_hosts
-          @hosts = associator.hosts
-          render 'api/v2/hosts/index', :layout => 'api/v2/layouts/index_layout'
+          handle_errors(associator)
         else
           render_message(_('Associating VMs is not supported for this compute resource'), :status => :unprocessable_entity)
         end
       end
 
+      api :PUT, "/compute_resources/:id/associate_vm/:vm_id", N_("Associate VM to Hosts")
+      param :id, :identifier, :required => true
+      param :vm_id, :identifier, :required => true
```

Comment:
The params should have descriptions, but no strong opinion here since the endpoint is pretty clear.